### PR TITLE
update `eslint-plugin-shopify` v32x

### PR DESCRIPTION
**Pull request by cartogram**
## Description

This PR updates `eslint-plugin-shopify` to version 32x. The main changes here are the grouping of imports and [requiring top-level describe](https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/require-top-level-describe.md) when using jest hooks in tests. Otherwise this is a really straightforward bump.

### CODE REVIEW

File: packages/dates/src/tests/is-less-than-one-minute-ago.test.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {clock} from '@shopify/jest-dom-mocks';
+
```
Comment (lemonmade) [inline packages/dates/src/tests/is-less-than-one-minute-ago.test.ts]: Hmm I am pretty surprised we are enforcing this.

File: test/each-test.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ if (typeof window !== 'undefined') {
   addClosest(window);
 }
 
-afterEach(() => {
-  destroyAll();
+describe('destroyAll()', () => {
```
Comment (lemonmade) [inline test/each-test.ts]: No, this is wrong. This was intended to be a global `afterEach` hook.

Comment (cartogram) [reply test/each-test.ts]: Im going to revert it and ignore the rule.

File: packages/dates/src/tests/is-less-than-one-minute-ago.test.ts (hunk 1 above)
Comment (cartogram) [reply packages/dates/src/tests/is-less-than-one-minute-ago.test.ts]: Its not something I would have added, but if people want it I think its fine.

**Pull request merged**